timeseries/src/autogluon/timeseries/dataset/ts_dataframe.py

```diff
@@ -3,7 +3,9 @@
 import copy
```

Comment:
This class seems to be getting a little entangled with dependencies on how it constructs things. Now looks like a good time to overhaul this. Specifically, 
- `from_` constructors seem to be able to call `__init__`. 
- Everyone seems to be able to call internal `_construct_*` methods, and `load_pd`
- `from_` constructors seem to be able to call each other.

This has also resulted in some redundancies such as 
- `_load_data_frame_from_file`.. which seems to have only one caller and its use appears redundant.
- `from_iterable` looks like it may not need a call to `_construct_.._from_iterable` as the call to `cls(...)` would anyway handle that. 

Normally `from_` constructors are there to explicitly prevent us from overloading the constructor. If we're allowing the constructor to take `Any`, then `from_` is not necessary unless for reasons for backward compatibility. This is why maybe we could redesign to ensure `from_` constructors just wrap `__init__` for now, and don't call private methods.

Reply:
Good idea, I've implemented this change.

Now `TimeSeriesDataFrame` accepts `data` in `DataFrame` / `str` / `Path` / `Iterable` formats. `from_*` methods are just thin wrappers that call `TimeSeriesDataFrame.__init__` and don't use internal methods.